scripts/rofl-puzzles.lic

```diff
@@ -0,0 +1,851 @@
+    end
+
+    def self.requirements
+      [ "white paper box" ]
+    end
+  end
+
+  module Candle
+    SUCCESS_REGEX = /The wick flickers gently but burns brightly/
+    FAIL_REGEX = /The wick refuses to stay lit/
+    ERROR_REGEX = /What were you referring to?/
+
+    def self.solve_puzzle
+      fput "look bench"
+      fput "get twig"
+      fput "put twig in brazier"
+      ["first", "second", "third", "fourth"].each do |ord|
+        case dothistimeout "light #{ord} candle", 3, Regexp.union(SUCCESS_REGEX, FAIL_REGEX, ERROR_REGEX)
+        when ERROR_REGEX, nil
+          respond
+          respond "  ERROR: this doesn't seem to be a candle puzzle"
+          respond
+          return
+        when SUCCESS_REGEX
+          break
+        end
+      end
+    end
+
+    def self.requirements
+      [
+        "iron brazier",
+        "elaborate prayer bench",
+      ]
+    end
+  end
+
+  Puzzle = Struct.new(:name, :klass, :description) do
+    def solve_puzzle
+      klass.solve_puzzle
+    end
+
+    def requirements
+      klass.requirements
+    end
+  end
+
+  PUZZLES = [
+    Puzzle.new("statue", Statue, "move some some statues around on an altar"),
+    Puzzle.new("mosaic", Mosaic, "find the right gem and symbol to put into the mosaic"),
+    Puzzle.new("crystal", CrystalPainting, "choose the correct damage type to destroy monster paintings"),
+    Puzzle.new("stars", ConstellationStars, "select the correct constellation to place in the night sky"),
+    Puzzle.new("ghost", Ghost, "use a wand to do stuff to a ghost"),
+    Puzzle.new("scramble", Scramble, "guess the word from scrambled letters"),
+    Puzzle.new("symbol", SymbolPainting, "paint an Arkati symbol the correct colors and assemble"),
+    Puzzle.new("box", PaperBox, "get some stuff from a box and do some things"),
+    Puzzle.new("candle", Candle, "light the correct candle"),
+  ]
+
+  def self.determine_puzzle
+    PUZZLES.find do |puzzle|
+      puzzle.requirements.all? do |requirement|
+        GameObj.loot.any? do |loot|
+          loot.noun == requirement || loot.name == requirement
+        end
+      end
+    end
+  end
+
+  def self.activate_puzzle_triggers
+    [
+      "white tile",
+      "tile",
+      "dark ceiling",
+    ].find do |trigger|
+      fput "look #{trigger}" if GameObj.loot.find { |i| i.name == trigger }
+    end
+  end
+
+  def self.puzzle_here?
+    activate_puzzle_triggers
+    determine_puzzle
+  end
+end
+
+print_usage_info = proc do
+  respond
+  respond "  Rings of Lumnis Puzzle Solver"
+  respond
+  respond "  Usage:"
+  respond
+  respond "    #{$clean_lich_char}#{script.name} <puzzle name>"
+  respond
+  respond "  where <puzzle name> is one of the following:"
+  respond
+  respond "    %-10s %s" % ["auto", "automatically solve puzzles when they appear (default behavior)"]
+  respond
+  RofL::PUZZLES.each do |puzzle|
+    respond "    %-10s %s" % [puzzle.name, puzzle.description]
+  end
+  respond
+end
+
+if defined?(Lich)
```

Comment:
This is my stupid ruby trick to facilitate testing. I can `load "./rofl-puzzles.lic` from the test file and it won't execute any of the code in here. I will probably make a separate issue about testing in general because I like to write specs for stuff when possible, but I also realize that almost no one else cares to do so.

Reply:
I love testing, and have plans for creating a mock CI environment we can use, just not time to implement it right now.  Maybe i can just sketch my ideas out in an issue